prowler/providers/aws/services/directoryservice/directoryservice_service.py

```diff
@@ -117,21 +118,27 @@ def __describe_event_topics__(self, regional_client):
         try:
             for directory in self.directories.values():
                 if directory.region == regional_client.region:
-                    describe_event_topics_parameters = {"DirectoryId": directory.id}
-                    event_topics = []
-                    describe_event_topics = regional_client.describe_event_topics(
-                        **describe_event_topics_parameters
-                    )
-                    for event_topic in describe_event_topics["EventTopics"]:
-                        event_topics.append(
-                            EventTopics(
-                                topic_arn=event_topic["TopicArn"],
-                                topic_name=event_topic["TopicName"],
-                                status=event_topic["Status"],
-                                created_date_time=event_topic["CreatedDateTime"],
+                    try:
+                        describe_event_topics_parameters = {"DirectoryId": directory.id}
+                        event_topics = []
+                        describe_event_topics = regional_client.describe_event_topics(
+                            **describe_event_topics_parameters
+                        )
+                        for event_topic in describe_event_topics["EventTopics"]:
+                            event_topics.append(
+                                EventTopics(
+                                    topic_arn=event_topic["TopicArn"],
+                                    topic_name=event_topic["TopicName"],
+                                    status=event_topic["Status"],
+                                    created_date_time=event_topic["CreatedDateTime"],
+                                )
                             )
+                        self.directories[directory.id].event_topics = event_topics
+                    except ClientError as error:
```

Comment:
Are we going to handle all `ClientError` this way? Should it be? 

The error was the following `ClientException[122]: An error occurred (ClientException) when calling the DescribeEventTopics operation: Operation is not supported for Shared MicrosoftAD directories.`

What if we look for `Operation is not supported for Shared MicrosoftAD directories.`? Also we have the `MicrosoftAD` type so we can play with that.

Reply:
Done